src/renderer/downloadsDropdown.tsx

```diff
@@ -74,7 +74,13 @@ class DownloadsDropdown extends React.PureComponent<Record<string, never>, State
     }
 
     clearAll = () => {
-        window.postMessage({type: REQUEST_CLEAR_DOWNLOADS_DROPDOWN}, window.location.href);
+        if (!this.clearAllButtonDisabled()) {
+            window.postMessage({type: REQUEST_CLEAR_DOWNLOADS_DROPDOWN}, window.location.href);
+        }
+    }
+
+    clearAllButtonDisabled = () => {
+        return this.state.downloads?.length === 1 && this.state.downloads[0]?.type === 'update';
```

Comment:
(clarification) So, based on the behaviour of the code above (clearDownloadsDropdown function), clearing when there is an update and other downloads, it will remove everything and push into the downloads the update again right?

Also, is it guaranteed that the update will always be the first of the downloads or something like `this.state.downloads.filter(d => d.type === 'update'). length > 0` or `this.state.downloads.map(d => d.type).includes('update')` is safer?